### PR TITLE
Passes default locale setting when executing commands.

### LINKED ISSUE #1347: runNative locale issues on linux
This code will not work as expected if locale in not english or C
https://github.com/oshi/oshi/blob/master/oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxCentralProcessor.java#L123

Proposed fix: set env. variable `LC_ALL=C`
I can't make PR because i don't have sane internet right now.

### PR DESCRIPTION
**Pull request by dmitraver**
Closes #1347 

Passes locale when executing commands on different platforms.

### CODE REVIEW

File: oshi-core/src/main/java/oshi/util/ExecutingCommand.java [hunk 1]
```diff
@@ -81,7 +83,16 @@ private ExecutingCommand() {
...
         try {
-            p = Runtime.getRuntime().exec(cmdToRunWithArgs);
+            PlatformEnum platform = SystemInfo.getCurrentPlatformEnum();
+            if (platform == PlatformEnum.WINDOWS) {
+                p = Runtime.getRuntime().exec(cmdToRunWithArgs, new String[]{"LANGUAGE=C"});
+            } else if (platform == PlatformEnum.LINUX || platform == PlatformEnum.FREEBSD
+                    || platform == PlatformEnum.SOLARIS || platform == PlatformEnum.AIX
+                    || platform == PlatformEnum.MACOSX) {
+                p = Runtime.getRuntime().exec(cmdToRunWithArgs, new String[]{"LC_ALL=C"});
+            } else {
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/util/ExecutingCommand.java:93]: This `else` case will never happen. It's either Windows or *nix. You can simplify the previous condition to be the "else" without testing the platform.

File: oshi-core/src/main/java/oshi/util/ExecutingCommand.java [hunk 2]
```diff
@@ -81,7 +83,16 @@ private ExecutingCommand() {
     public static List<String> runNative(String[] cmdToRunWithArgs) {
         Process p = null;
         try {
-            p = Runtime.getRuntime().exec(cmdToRunWithArgs);
+            PlatformEnum platform = SystemInfo.getCurrentPlatformEnum();
+            if (platform == PlatformEnum.WINDOWS) {
+                p = Runtime.getRuntime().exec(cmdToRunWithArgs, new String[]{"LANGUAGE=C"});
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/util/ExecutingCommand.java:88]: Let's make this string array a constant so we don't have to instantiate it every time.

File: oshi-core/src/main/java/oshi/util/ExecutingCommand.java [hunk 3]
```diff
@@ -81,7 +83,16 @@ private ExecutingCommand() {
...
+            } else if (platform == PlatformEnum.LINUX || platform == PlatformEnum.FREEBSD
+                    || platform == PlatformEnum.SOLARIS || platform == PlatformEnum.AIX
+                    || platform == PlatformEnum.MACOSX) {
+                p = Runtime.getRuntime().exec(cmdToRunWithArgs, new String[]{"LC_ALL=C"});
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/util/ExecutingCommand.java:92]: Let's make this string array a constant too.  Actually you can probably just define a single constant `EXEC_ENV` (or similar) and use the platform conditional to decide which array it is. The platform won't change during execution.  Then you can just use that as the second arg here with no conditional.

**Review by dbwiddis — CHANGES_REQUESTED** [review]
Thanks. This works but I've left some comments on improving it.

Can you also add a change log entry under 5.3.0?

**Comment by XakepSDK** [comment]
Maybe helper method? `ENVP_WINDOWS` and `ENVP_UNIX` constants.
```java
private static String[] getDefaultEnv() {
    if (platform == PlatformEnum.WINDOWS) {
        return ENVP_WINDOWS;
    } else if (platform != PlatformEnum.UNKNOWN) {
        return ENVP_UNIX;
    } else {
        return null;
    }
}
```

**Comment by dbwiddis** [comment]
> Maybe helper method?

In the direction I was thinking but this will not change during operation, so we just need a one-time constant (So using this helper method but defining a constant as the result would work).

File: oshi-core/src/main/java/oshi/util/ExecutingCommand.java (hunk 1 above)
Comment (dmitraver) [reply oshi-core/src/main/java/oshi/util/ExecutingCommand.java:93]: I actually thought about checking for windows and treat everything else as *nix but then found that there is also UNKNOWN value in the enum so wanted to make sure that there are no surprises when new not windows and not unix platform is added but if you say its fine I will change it 👍

**Comment by dmitraver** [comment]
Pushed review fixes.

**Comment by dmitraver** [comment]
Ops, haven't seen your comment about changelog, adding it.

**Comment by dmitraver** [comment]
@dbwiddis updated changelog and rebased.

**Comment by dbwiddis** [comment]
This looks good!  I'm headed to bed but will get this merged tomorrow. Thanks for your contribution!

File: oshi-core/src/main/java/oshi/util/ExecutingCommand.java (hunk 1 above)
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/ExecutingCommand.java]: It is in the enum, but also throws an exception at SystemInfo instantiation.

**Pull request merged**